src/test/scala/com/cognite/sdk/scala/common/DataModelInstancesSerializerTest.scala

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Cognite AS
+// SPDX-License-Identifier: Apache-2.0
+
+package com.cognite.sdk.scala.common
+
+import com.cognite.sdk.scala.v1.{ArrayProperty, BooleanProperty, NumberProperty, PropertyType, StringProperty}
+import io.circe.generic.semiauto.deriveDecoder
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+final case class DMIResponse(
+    modelExternalId: String,
+    properties: Option[Map[String, PropertyType]] = None
+)
+
+@SuppressWarnings(
+  Array(
+    "org.wartremover.warts.JavaSerializable",
+    "org.wartremover.warts.Serializable",
+    "org.wartremover.warts.NonUnitStatements",
+    "org.wartremover.warts.Product"
+  )
+)
+class DataModelInstancesSerializerTest extends AnyWordSpec with Matchers {
+
+  import cats.syntax.functor._
+  import io.circe.Decoder
+  import io.circe.parser.decode
+
+  implicit val dmiResponseDecoder: Decoder[DMIResponse] =
+    deriveDecoder[DMIResponse]
+
+  implicit val decodeProp: Decoder[PropertyType] =
```

Comment:
I add a test for empty array, look like when it's empty it doesn't care about the type though (i.e  `ArrayProperty(Vector())` is doing just fine) I will see if we can combine this with retrieving with the actual schema, maybe can do smthing lie this 🤔 
```
prop match {
    case "boolean" => Decoder.decodeBoolean.map(BooleanProperty(_)).widen
    case "text" => Decoder.decodeString.map(StringProperty(_)).widen
    case "float64[]" |  "float32[]" => Decoder
        .decodeArray[Double]
        .map(x => ArrayProperty[NumberProperty](x.toVector.map(NumberProperty(_))))
        .widen
    ....
}
```

Reply:
Yeah, but it's not true that it's "any type" if it's really `float64[]`. Also, if we do have `float32[]` and `float64[]` they should be treated differently, as `Float` and `Double`. Not to mention the issues with `date` being sent as a string, etc.

And yes, that's the right idea with using the schema, creating a function from the schema to a decoder. I think this could be ok as an intermediate step, if we need it now, but I don't think we can avoid using the schema.